File: plugins/nend/proj.android/src/org/cocos2dx/plugin/AdsNend.java
/****************************************************************************
Copyright (c) 2012-2013 cocos2d-x.org

http://www.cocos2d-x.org

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
THE SOFTWARE.
****************************************************************************/
package org.cocos2dx.plugin;

import java.util.Hashtable;

import android.app.Activity;
import android.content.Context;
import android.util.Log;
import android.view.WindowManager;

import net.nend.android.NendAdView;

public class AdsNend implements InterfaceAds {

    private static final String LOG_TAG = "AdsNend";
    private static Activity mContext = null;
    private static boolean bDebug = false;
    private static AdsNend mAdapter = null;

    private NendAdView adView = null;
    private WindowManager mWm = null;

    protected static void LogE(String msg, Exception e) {
        Log.e(LOG_TAG, msg, e);
        e.printStackTrace();
    }

    protected static void LogD(String msg) {
        if (bDebug) {
            Log.d(LOG_TAG, msg);
        }
    }

    public AdsNend(Context context) {
        mContext = (Activity) context;
        mAdapter = this;
    }

    @Override
    public void setDebugMode(boolean debug) {
        bDebug = debug;
    }

    @Override
    public void configDeveloperInfo(Hashtable<String, String> devInfo) {
        try {
            String apiKey = devInfo.get("ApiKey");
            String spotId = devInfo.get("SpotId");
            LogD("init ids ApiKey:" + apiKey + ", SpotId:" + spotId);
            adView = new NendAdView(mContext, Integer.parseInt(spotId), apiKey);
            adView.loadAd();
        } catch (Exception e) {
            LogE("initAppInfo, The format of appInfo is wrong", e);
        }
    }

    @Override
    public void showAds(Hashtable<String, String> info, final int pos) {
        PluginWrapper.runOnMainThread(new Runnable() {
            @Override
            public void run() {
                if (null == mWm) {
                    mWm = (WindowManager) mContext.getSystemService("window");
                }
                AdsWrapper.addAdView(mWm, adView, pos);
            }
        });
    }

    @Override
    public void hideAds(Hashtable<String, String> info) {
        PluginWrapper.runOnMainThread(new Runnable() {
            @Override
            public void run() {
                if (null != adView && null != mWm) {
                    mWm.removeView(adView);
                }
            }
        });
    }

    @Override
    public String getPluginVersion() {
        return "0.0.1";
    }

    @Override
    public String getSDKVersion() {
        return "2.4.1";
    }

    @Override
    public void queryPoints() {
        LogD("Nend not support query points!");
    }

    @Override
    public void spendPoints(int points) {
        LogD("Nend not support spend points!");
    }

}

File: plugins/nend/proj.ios/AdsNend.m
/****************************************************************************
 Copyright (c) 2013 cocos2d-x.org
 
 http://www.cocos2d-x.org
 
 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:
 
 The above copyright notice and this permission notice shall be included in
 all copies or substantial portions of the Software.
 
 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 ****************************************************************************/

#import "AdsNend.h"
#import "AdsWrapper.h"

#define OUTPUT_LOG(...)     if (self.debug) NSLog(__VA_ARGS__);

@implementation AdsNend

@synthesize debug = __debug;

- (void) dealloc
{
    if (self.bannerView != nil) {
        [self.bannerView setDelegate:nil];
        [self.bannerView release];
        self.bannerView = nil;
    }
    [super dealloc];
}

#pragma mark InterfaceAds impl

- (void) configDeveloperInfo: (NSMutableDictionary*) devInfo
{
    NSString* apiKey = (NSString*)[devInfo objectForKey:@"ApiKey"];
    NSString* spotId = (NSString*)[devInfo objectForKey:@"SpotId"];
    self.bannerView = [[NADView alloc] initWithFrame:CGRectMake(0, 0, 320, 50)];
    [self.bannerView setIsOutputLog:self.debug];
    [self.bannerView setNendID:apiKey spotID:spotId];
    [self.bannerView setDelegate:self];
    [self.bannerView load];
}

- (void) showAds: (NSMutableDictionary*) info position:(int) pos
{
    [AdsWrapper addAdView:self.bannerView atPos:pos];
}

- (void) hideAds: (NSMutableDictionary*) info
{
    [self.bannerView removeFromSuperview];
}

- (void) queryPoints
{
    OUTPUT_LOG(@"Nend not support query points!");
}

- (void) spendPoints: (int) points
{
    OUTPUT_LOG(@"Nend not support spend points!");
}

- (void) setDebugMode: (BOOL) isDebugMode
{
    self.debug = isDebugMode;
    if (self.bannerView) {
        [self.bannerView setIsOutputLog:self.debug];
    }
}

- (NSString*) getSDKVersion
{
    return @"2.5.5";
}

- (NSString*) getPluginVersion
{
    return @"0.0.1";
}

- (void)nadViewDidFinishLoad:(NADView *)adView
{
}

- (void)nadViewDidClickAd:(NADView *)adView
{
    [AdsWrapper onAdsResult:self withRet:kAdsShown withMsg:@"nend"];
}

- (void)nadViewDidReceiveAd:(NADView *)adView
{
}

- (void)nadViewDidFailToReceiveAd:(NADView *)adView
{
    OUTPUT_LOG(@"delegate nadViewDidFailToLoad:");
    NSError* error = adView.error;
    NSString* domain = error.domain;
    int errorCode = error.code;
    OUTPUT_LOG(@"%@",[NSString stringWithFormat: @"code=%d, message=%@", errorCode, domain]);
    switch (errorCode) {
        case NADVIEW_AD_SIZE_TOO_LARGE:
            OUTPUT_LOG(@"広告サイズがディスプレイサイズよりも大きい");
            break;
        case NADVIEW_INVALID_RESPONSE_TYPE:
            OUTPUT_LOG(@"不明な広告ビュータイプ");
            break;
        case NADVIEW_FAILED_AD_REQUEST:
            OUTPUT_LOG(@"広告取得失敗");
            break;
        case NADVIEW_FAILED_AD_DOWNLOAD:
            OUTPUT_LOG(@"広告画像の取得失敗");
            break;
        case NADVIEW_AD_SIZE_DIFFERENCES:
            OUTPUT_LOG(@"リクエストしたサイズと取得したサイズが異なる");
            break;
        default:
            break;
    }
}

@end
